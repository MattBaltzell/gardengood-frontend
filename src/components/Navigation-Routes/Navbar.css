.Navbar {
  background-color: var(--color-neutral-600);
  padding: 1.2rem;
  width: 100%;
  display: flex;
  justify-content: center;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2);
  overflow: hidden;
}

.Navbar__container {
  width: 100%;
  max-width: 1200px;
}

.Navbar__container nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.Navbar__logo {
  height: 2.2rem;
}

a.Navbar__logo-link {
  display: flex;
}

.Navbar__links {
  display: flex;
  gap: 1.5rem;
}

.Navbar__links a {
  text-decoration: none;
  color: var(--color-neutral-200);
  position: relative;
}

.Navbar__links a::after {
  content: "";
  height: 3px;
  width: 0;
  opacity: 0;
  margin: 0 auto;
  background-color: var(--color-primary-400);
  bottom: -0.6rem;
  left: 0;
  right: 0;
  transition: all 0.3s ease-in-out;
}

.Navbar__links a.active {
  color: var(--color-neutral-100);
}

.Navbar__links a.active::after {
  width: 100%;
  opacity: 1;
  position: absolute;
}

.Navbar__links a:hover {
  color: var(--color-neutral-100);
}

.Navbar__links a:hover::after {
  width: 100%;
  transform: translateX(0);
  opacity: 1;
  position: absolute;
}
