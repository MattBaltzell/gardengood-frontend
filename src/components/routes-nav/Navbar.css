.Navbar {
  background-color: var(--color-neutral-600);
  z-index: 1000;

  padding: 1.2rem 1rem;
  width: 100%;
  display: flex;
  justify-content: center;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2);
}

.Navbar__container {
  width: 100%;
  max-width: 1200px;
}

.Navbar__container nav {
  display: flex;
  gap: 2rem;
  justify-content: space-between;
  align-items: center;
  position: relative;
}

.Navbar__constants {
  display: flex;
  justify-content: space-between;
  width: 100%;
}

.Navbar__logo {
  height: 2.4em;
}

a.Navbar__logo-link {
  display: flex;
}

.Navbar__weather {
  align-items: center;
  display: flex;
  gap: 1rem;
}
.Navbar__weather-condition {
  align-items: center;
  display: flex;
}
.Navbar__weather-icon {
  width: 2.4rem;
}

.Navbar__menu-icon {
  cursor: pointer;
  display: flex;
  flex-direction: column;
  gap: 5px;
}

.Navbar__menu-icon div {
  height: 3px;
  width: 2rem;
  background-color: var(--color-neutral-200);
  z-index: 1000;
  transition: all 0.2s ease-out;
}

.Navbar__menu-icon.active > div:first-child {
  transform: translateX(0px) translateY(8px) rotate(135deg);
}

.Navbar__menu-icon.active > div:nth-child(2) {
  opacity: 0;
}

.Navbar__menu-icon.active > div:last-child {
  transform: translateX(0px) translateY(-8px) rotate(-135deg);
}

.Navbar__menu {
  display: flex;
  align-items: center;
  justify-content: center;
  position: fixed;
  top: 76.77px;
  right: -100vw;
  padding: 2rem;
  border-radius: 0 0 0 4px;
  min-width: 50vw;
  background-color: var(--color-primary-400);
  transition: all 0.3s ease-in-out;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2);
}

.Navbar__menu.active {
  right: 0;
}

.Navbar__links {
  display: flex;
  flex-direction: column;
  width: 100%;
  font-size: 1.2rem;
  gap: 1rem;
  align-items: flex-start;
  white-space: nowrap;
}

.Navbar__links a {
  text-decoration: none;
  color: var(--color-neutral-100);
  transition: all 0.3s ease-in-out;
  position: relative;
}

.Navbar__links a::after {
  content: "";
  height: 3px;
  width: 0;
  opacity: 0;
  margin: 0 auto;
  background-color: var(--color-neutral-700);
  bottom: -0.6rem;
  left: 0;
  right: 0;
  transition: all 0.3s ease-in-out;
}

.Navbar__links a.active {
  color: var(--color-neutral-700);
}

.Navbar__links a.active::after {
  width: 100%;
  opacity: 1;
  position: absolute;
}

.Navbar__links a:hover {
  color: var(--color-neutral-700);
}

.Navbar__links a:hover::after {
  width: 100%;
  transform: translateX(0);
  opacity: 1;
  position: absolute;
}

@media screen and (min-width: 800px) {
  .Navbar {
    padding: 1.2rem 3rem;
  }

  /* .Navbar__menu {
    display: flex;
    align-items: center;
    justify-content: stretch;
    position: static;
    padding: 0;
    top: auto;
    right: auto;
    background-color: transparent;
    min-width: auto;
    box-shadow: none;
  }
  .Navbar__links {
    flex-direction: row;
    gap: 2.5rem;
    font-size: 1rem;
  }

  .Navbar__menu-icon {
    display: none;
  }

  .Navbar__links a::after {
    background-color: var(--color-primary-400);
  }

  .Navbar__links a.active {
    color: var(--color-neutral-100);
  }

  .Navbar__links a:hover {
    color: var(--color-primary-200);
  } */
}
