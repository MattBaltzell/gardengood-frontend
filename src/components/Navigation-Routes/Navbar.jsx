import "./Navbar.css";
import gardenGoodLogoMark from "../../GardenGood-logomark.svg";
import { NavLink, Link, useNavigate } from "react-router-dom";
import UserContext from "../../auth/UserContext";
import WeatherContext from "../../WeatherContext";
import { useContext, useState, useRef } from "react";

const Navbar = ({ logout }) => {
  const user = useContext(UserContext);
  const weather = useContext(WeatherContext);
  const [showNavbar, setShowNavbar] = useState(false);
  const navigate = useNavigate();

  const handleShowNavbar = () => {
    setShowNavbar(!showNavbar);
  };

  const handleLogout = async () => {
    await logout();
    setShowNavbar(false);
    navigate("/");
  };

  return (
    <>
      <div className="Navbar">
        <div className="Navbar__container">
          <nav>
            <div className="Navbar__constants">
              <NavLink to="/" className="Navbar__logo-link">
                <img
                  className="Navbar__logo"
                  src={gardenGoodLogoMark}
                  alt="GardenGood"
                />
              </NavLink>

              {user && weather ? (
                <div className="Navbar__weather">
                  <div className="Navbar__weather-condition">
                    <img
                      className="Navbar__weather-icon"
                      src={weather.condition.icon}
                      alt="Weather condition"
                    />
                    <p>{weather.condition.text}</p>
                  </div>
                  <p>{Math.round(weather.temp_f)} FÂ°</p>
                </div>
              ) : null}
            </div>

            <div className={`Navbar__menu ${showNavbar ? "active" : ""}`}>
              {!user ? (
                <div className="Navbar__links">
                  <NavLink to="/login">Login</NavLink>
                  <NavLink to="/Signup">Signup</NavLink>
                </div>
              ) : (
                <div className="Navbar__links">
                  <NavLink to="/plants">Plants</NavLink>
                  <Link onClick={handleLogout}>Logout {user.username}</Link>
                </div>
              )}
            </div>
            <div
              onClick={handleShowNavbar}
              className={`Navbar__menu-icon ${showNavbar ? "active" : ""}`}
            >
              <div></div>
              <div></div>
              <div></div>
            </div>
          </nav>
        </div>
      </div>
    </>
  );
};

export default Navbar;
