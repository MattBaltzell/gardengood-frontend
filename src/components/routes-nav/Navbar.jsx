import "./Navbar.css";
import gardenGoodLogoMark from "../../GardenGood-logomark.svg";
import { NavLink, Link, useNavigate } from "react-router-dom";
import UserContext from "../auth/UserContext";
import WeatherContext from "../../WeatherContext";
import { useContext, useEffect, useRef } from "react";

const Navbar = ({ logout, menuIsOpen, handleMenuIsOpen }) => {
  const user = useContext(UserContext);
  const weather = useContext(WeatherContext);
  const navigate = useNavigate();
  const menuRef = useRef();

  useEffect(() => {
    const checkIfClickedOutside = (e) => {
      // If the menu is open and the clicked target is not within the menu,
      // then close the menu

      if (
        menuIsOpen &&
        menuRef.current &&
        !e.target.classList.contains("Navbar__menu-icon-bar") &&
        !menuRef.current.contains(e.target)
      ) {
        handleMenuIsOpen();
      }
    };

    document.addEventListener("mouseup", checkIfClickedOutside);

    return () => {
      // Cleanup the event listener
      document.removeEventListener("mouseup", checkIfClickedOutside);
    };
  }, [menuIsOpen]);

  const handleLogout = async () => {
    await logout();
    navigate("/");
  };

  return (
    <>
      <div className="Navbar">
        <div className="Navbar__container">
          <nav>
            <NavLink to="/" className="Navbar__logo-link">
              <img
                className="Navbar__logo"
                src={gardenGoodLogoMark}
                alt="GardenGood"
              />
            </NavLink>

            {user && weather ? (
              <div className="Navbar__weather">
                <div className="Navbar__weather-condition">
                  <img
                    className="Navbar__weather-icon"
                    src={weather.condition.icon}
                    alt="Weather condition"
                  />
                  <p>{weather.condition.text}</p>
                </div>
                <p>{Math.round(weather.temp_f)} FÂ°</p>
              </div>
            ) : null}

            <div
              onClick={handleMenuIsOpen}
              className={`Navbar__menu-icon ${menuIsOpen ? "active" : ""}`}
            >
              <div className="Navbar__menu-icon-bar"></div>
              <div className="Navbar__menu-icon-bar"></div>
              <div className="Navbar__menu-icon-bar"></div>
            </div>

            <div
              ref={menuRef}
              className={`Navbar__menu ${menuIsOpen ? "active" : ""}`}
            >
              {!user ? (
                <div className="Navbar__links">
                  <NavLink to="/login" onClick={handleMenuIsOpen}>
                    Login
                  </NavLink>
                  <NavLink to="/Signup" onClick={handleMenuIsOpen}>
                    Signup
                  </NavLink>
                </div>
              ) : (
                <div className="Navbar__links">
                  <NavLink to="/plants" onClick={handleMenuIsOpen}>
                    Plants
                  </NavLink>
                  <NavLink
                    to={`/users/${user.username}`}
                    onClick={handleMenuIsOpen}
                  >
                    {`${
                      user.username[0].toUpperCase() + user.username.slice(1)
                    }'s Account`}
                  </NavLink>
                  <Link onClick={handleLogout}>Logout</Link>
                </div>
              )}
            </div>
          </nav>
        </div>
      </div>
    </>
  );
};

export default Navbar;
