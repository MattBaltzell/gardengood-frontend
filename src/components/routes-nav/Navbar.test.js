import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";

import Navbar from "./Navbar";
import UserContext from "../auth/UserContext";

const user = {
  username: "testuser",
  first_name: "firstname",
  last_name: "lastname",
  email: "test@test.com",
};

describe("Navbar component tests", function () {
  it("renders without crashing", function () {
    render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    );
  });

  it("matches snapshot", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <UserContext.Provider value={user}>
          <Navbar />
        </UserContext.Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot when logged out", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <UserContext.Provider value={null}>
          <Navbar />
        </UserContext.Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
