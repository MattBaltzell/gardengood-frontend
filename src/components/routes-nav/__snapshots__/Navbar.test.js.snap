// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component tests matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Navbar"
  >
    <div
      class="Navbar__container"
    >
      <nav>
        <a
          aria-current="page"
          class="Navbar__logo-link active"
          href="/"
        >
          <img
            alt="GardenGood"
            class="Navbar__logo"
            src="GardenGood-logomark.svg"
          />
        </a>
        <div
          class="Navbar__menu-icon "
          data-testid="hamburger-icon"
        >
          <div
            class="Navbar__menu-icon-bar"
          />
          <div
            class="Navbar__menu-icon-bar"
          />
          <div
            class="Navbar__menu-icon-bar"
          />
        </div>
        <div
          class="Navbar__menu "
          data-testid="nav-menu"
        >
          <div
            class="Navbar__links"
          >
            <a
              class=""
              href="/plants"
            >
              Plant Directory
            </a>
            <a
              class=""
              href="/users/testuser"
            >
              Testuser's Account
            </a>
            <a
              href="/"
            >
              Logout
            </a>
          </div>
        </div>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Navbar component tests matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="Navbar"
  >
    <div
      class="Navbar__container"
    >
      <nav>
        <a
          aria-current="page"
          class="Navbar__logo-link active"
          href="/"
        >
          <img
            alt="GardenGood"
            class="Navbar__logo"
            src="GardenGood-logomark.svg"
          />
        </a>
        <div
          class="Navbar__menu-icon "
          data-testid="hamburger-icon"
        >
          <div
            class="Navbar__menu-icon-bar"
          />
          <div
            class="Navbar__menu-icon-bar"
          />
          <div
            class="Navbar__menu-icon-bar"
          />
        </div>
        <div
          class="Navbar__menu "
          data-testid="nav-menu"
        >
          <div
            class="Navbar__links"
          >
            <a
              class=""
              href="/login"
            >
              Login
            </a>
            <a
              class=""
              href="/Signup"
            >
              Signup
            </a>
          </div>
        </div>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;
